Example-UIKit/Example-UIKit/System/SideBarViewController.swift

```diff
@@ -0,0 +1,178 @@
+        }
+
+        collectionView = UICollectionView(frame: view.bounds, collectionViewLayout: layout)
+        collectionView.autoresizingMask = [.flexibleWidth, .flexibleHeight]
+        collectionView.delegate = self
+        view.addSubview(collectionView)
+    }
+
+    private func configureDataSource() {
+        let cellRegistration = UICollectionView.CellRegistration<UICollectionViewListCell, SidebarItem> {
+            (cell, indexPath, item) in
+            var content = cell.defaultContentConfiguration()
+            content.text = item.rawValue
+            cell.contentConfiguration = content
+        }
+
+        dataSource = UICollectionViewDiffableDataSource<Int, SidebarItem>(collectionView: collectionView) {
+            (collectionView: UICollectionView, indexPath: IndexPath, identifier: SidebarItem) -> UICollectionViewCell? in
+            return collectionView.dequeueConfiguredReusableCell(using: cellRegistration, for: indexPath, item: identifier)
+        }
+
+        var snapshot = NSDiffableDataSourceSnapshot<Int, SidebarItem>()
+        snapshot.appendSections([0])
+        snapshot.appendItems(SidebarItem.allCases)
+        dataSource.apply(snapshot, animatingDifferences: false)
+    }
+
+    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        let selectedItem = SidebarItem.allCases[indexPath.item]
+        delegate?.didSelectSidebarItem(selectedItem)
+    }
+}
+
+// MARK: - Sidebar Delegate
+extension SideBarViewController: SidebarContentDelegate {
+    func didSelectSidebarItem(_ item: SidebarItem) {
+        // Handle navigation based on selected item
+        let detailVC: UIViewController
+        
+        switch item {
+        case .home:
+            detailVC = UINavigationController(rootViewController: HomeViewController())
+        case .splitViewDemo:
+            detailVC = UINavigationController(rootViewController: SplitViewDemoViewController())
+        }
+
+        _splitViewController.setViewController(detailVC, for: .secondary)
+    }
+    
+    private func createDetailViewController(title: String, message: String) -> UIViewController {
+        let detailVC = UIViewController()
+        detailVC.view.backgroundColor = .secondarySystemBackground
+        detailVC.title = title
+        
+        let label = UILabel()
+        label.text = message
+        label.textAlignment = .center
+        label.numberOfLines = 0
+        label.translatesAutoresizingMaskIntoConstraints = false
+        detailVC.view.addSubview(label)
+        NSLayoutConstraint.activate([
+            label.centerXAnchor.constraint(equalTo: detailVC.view.centerXAnchor),
+            label.centerYAnchor.constraint(equalTo: detailVC.view.centerYAnchor),
+            label.leadingAnchor.constraint(greaterThanOrEqualTo: detailVC.view.leadingAnchor, constant: 20),
+            label.trailingAnchor.constraint(lessThanOrEqualTo: detailVC.view.trailingAnchor, constant: -20)
+        ])
```

Comment:
Just a quick heads up, you can use SnapKit here to make it less verbose but otherwise lgtm